cmd/gbounty/bootstrap/update.go

```diff
@@ -22,12 +23,56 @@ import (
 	"github.com/bountysecurity/gbounty/kit/slices"
 )
 
+func getLastCheckTime() (time.Time, error) {
+	filePath, err := lastCheckFilePath()
+	if err != nil {
+		return time.Time{}, err
+	}
+
+	data, err := os.ReadFile(filePath)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return time.Time{}, nil
+		}
+		return time.Time{}, err
+	}
+
+	return time.Parse(time.RFC3339, string(data))
+}
+
+func checkForUpdatesRequired() bool {
+	if slices.In(os.Args, "--update") || slices.In(os.Args, "--check-updates") {
+		return true
+	}
+
+	const updateInterval = 24 * time.Hour
+	lastCheckTime, err := getLastCheckTime()
+	currentTime := time.Now()
+
+	if err != nil || lastCheckTime.IsZero() || currentTime.Sub(lastCheckTime) > updateInterval {
+		return true
+	}
+	return false
+}
+
 func CheckForUpdates() {
-	// Disable checks on CI jobs.
+	// Disable checks on CI jobs and when updates not required.
 	if _, isCI := os.LookupEnv("CI"); isCI {
 		return
 	}
 
+	updatesRequired := checkForUpdatesRequired()
+	if !updatesRequired {
+		return
+	}
+
+	if slices.In(os.Args, "--update") || updatesRequired {
```

Comment:
Isn't the call to `slices.In` redundant here? 🤔

Reply:
sorry i think this is mistakenly. will modify this